dotnet/src/SemanticKernel.Core/Functions/InstrumentedSKFunction.cs

```diff
@@ -18,153 +18,96 @@
 namespace Microsoft.SemanticKernel;
 #pragma warning restore IDE0130
 
-/// <summary>
-/// Standard Semantic Kernel callable function with instrumentation.
-/// </summary>
+/// <summary>Instrumented function that surrounds the invocation of another function with loggic and metrics.</summary>
 internal sealed class InstrumentedSKFunction : ISKFunction
 {
-    /// <inheritdoc/>
-    public string Name => this._function.Name;
-
-    /// <inheritdoc/>
-    public string PluginName => this._function.PluginName;
-
-    /// <inheritdoc/>
-    public string Description => this._function.Description;
+    /// <summary><see cref="ActivitySource"/> for function-related activities.</summary>
+    private static readonly ActivitySource s_activitySource = new(typeof(ISKFunction).FullName!);
+
+    /// <summary><see cref="Meter"/> for function-related metrics.</summary>
+    private static readonly Meter s_meter = new(typeof(ISKFunction).FullName!);
+
+    /// <summary><see cref="Histogram{T}"/> to record function invocation duration.</summary>
+    private static readonly Histogram<double> s_invocationDuration = s_meter.CreateHistogram<double>(
+        name: "sk.function.duration",
+        unit: "s",
+        description: "Measures the duration of a function’s execution");
+
+    /// <summary><see cref="Counter{T}"/> to track the number of successful function invocations.</summary>
+    private static readonly Counter<int> s_successfulInvocations = s_meter.CreateCounter<int>(
+        name: "sk.function.invocations",
+        unit: "{invocation}",
+        description: "Number of successful function invocations");
+
+    /// <summary><see cref="Counter{T}"/> to track the number of failed function invocations.</summary>
+    private static readonly Counter<int> s_errors = s_meter.CreateCounter<int>(
+        name: "sk.function.errors",
+        unit: "{error}",
+        description: "Number of function invocation errors");
```

Comment:
You tell me :-)

I was following examples from opentelemetry, like for FaaS. What do we do in similar situations?

Will Aspire show all the relevant information in the dashboard of we just use duration?

Reply:
> I was following examples from opentelemetry, like for FaaS. What do we do in similar situations?

OTEL semantic conventions for HTTP only have a duration histogram and active_requests up/down counter. It doesn't have total counters. The reason is that the histogram tracks values and counts.

> Will Aspire show all the relevant information in the dashboard if we just use duration? or if we just use invocations, will it easily show a chart of errors?

That is a good point. In Grafana you can access histogram values as counts and then create charts to display counts. Aspire always displays histograms as P50/P90/P99 charts without counts.

The problem here is Aspire. It should have the option to switch histograms between displaying percentiles and counts.

Reply:
Ok, I can consolidate this to just duration.

> That is a good point. [...] The problem here is Aspire. It should have the option to switch histograms between displaying percentiles and counts.

Sounds like you're on it?  And I assume it'll enable filtering / displaying based on tags / attributes so that you can see the difference between success / errors?

Reply:
I read https://github.com/dotnet/aspire/issues/965; I don't see any mention of tags / attributes.  Will someone be able to differentiate successes from failures in a view where the histogram is rendered as counts?

Reply:
Yes. It will have all the same filtering capability.